### PR TITLE
Implement error dialog on scanning for invalid signatures (EXPOSUREAPP-8367)

### PR DESCRIPTION
**Pull request by mtwalli**
After: https://github.com/corona-warn-app/cwa-app-android/pull/3779

Screenshot 2021-07-26 at 11 38 47

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/exception/InvalidHealthCertificateException.kt [hunk 1]
```diff
@@ -56,11 +56,7 @@ open class InvalidHealthCertificateException(
         HC_DSC_EXPIRED("DSC expired."),
     }
 
-    open val showFaqButton: Boolean
-        get() = errorCode in codesCertificateInvalid
-    open val faqButtonText: Int = R.string.error_button_dcc_faq
-    open val faqLink: Int = R.string.error_button_dcc_faq_link
-
+    val isCertificateInvalid: Boolean get() = errorCode in codesCertificateInvalid
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/exception/InvalidHealthCertificateException.kt:59]: nice boolean

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/exception/InvalidHealthCertificateException.kt [hunk 2]
```diff
@@ -75,6 +71,20 @@ open class InvalidHealthCertificateException(
         ErrorCode.HC_JSON_SCHEMA_INVALID
     )
 
+    val isSignatureInvalid: Boolean get() = errorCode in signatureErrorCodes
+    private val signatureErrorCodes = listOf(
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/exception/InvalidHealthCertificateException.kt:75]: nice List

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/exception/InvalidHealthCertificateException.kt [hunk 3]
```diff
@@ -98,6 +112,7 @@ open class InvalidHealthCertificateException(
     }
 }
 
+private const val ERROR_MESSAGE_SIGNATURE_INVALID = R.string.dcc_signature_validation_dialog_message
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/exception/InvalidHealthCertificateException.kt:115]: NICE Error Message ;-)

**Review by BMItr — DISMISSED** [review]
lgtm!

**Comment by mtwalli** [comment]
> lgtm!

Nice review 😄

File: Corona-Warn-App/src/main/res/values-de/covid_certificate_strings.xml [hunk 4]
```diff
@@ -266,4 +266,16 @@
     <string name="expiration_date">Gültig bis %1$s</string>
     <!-- XHED: Vaccination Detail expiration info -->
     <string name="expiration_info">Bitte bemühen Sie sich rechtzeitig darum, einen neuen digitalen Nachweis ausstellen zu lassen.</string>
+
+    <!--    Dcc Signature validation-->
+    <!-- XHED: Signature validation dialog title-->
+    <string name="dcc_signature_validation_dialog_title">Signatur ungültig</string>
+    <!-- XTXT: Signature validation dialog message-->
+    <string name="dcc_signature_validation_dialog_message">Die Signatur des Zertifikats ist ungültig. Das Zertifikat kann nicht als Nachweis verwendet werden und daher nicht importiert werden</string>
```
Comment (ChristinaTsigganos) [inline Corona-Warn-App/src/main/res/values-de/covid_certificate_strings.xml:274]: please add a dot "." at the end of the sentence. Other than that, the PR is UA approved.

**Review by ChristinaTsigganos — CHANGES_REQUESTED** [review]
just minor issue, please add a period at end of sentence

File: Corona-Warn-App/src/main/res/values-de/covid_certificate_strings.xml (hunk 4 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/res/values-de/covid_certificate_strings.xml]: Done

**Review by ChristinaTsigganos — APPROVED** [review]
UA approved

**Review by d4rken — APPROVED** [review]
lgtm

**Pull request merged**